### PR TITLE
The CUM Update - First Iteration.

### PR DESCRIPTION
**Pull request by Memeji**
# Description

This update adds two traits (Cock, Boobs) and ways to make cum and milk using them.

---

# TODO

- [ ] Check for suit (except loincloth) to prevent action.
- [ ] Be able to cum/milk w/o a container in hand.
- [ ] Better text for actions.
- [ ] Clean the code
- [ ] "Pussy" trait and components with mechanics.

---

# Changelog

:cl:
- add: "Cock" trait, fill containers with your essence!
- add: "Boobs" trait, fill containers using your mommy milkers!
- add: LewdTrait system.
- tweak: Changed hunger/thirst values for cum to not satiate as much. (No cum meta)
- changed: "Penis" component to "CumProducer" component.
- changed: "Boobs" component to "MilkProducer" component.

### CODE REVIEW

**Review by FoxxoTrystan — COMMENTED** [review]
HasBoobs and HasCock as just the same system, just add 2 component but they use the same system to CockComponent use "LewdSystem" or i dunno how you wana call it but avoid using 2 Systems thats are mostly copy and paste, i recomend also using .yml so instead you also do only 1 Component and System just regrouping them under 1, look how languages are listed by example.

I will fully review this later so for now this is a comment and not requested changes.

**Comment by Fansana** [comment]
I'm unsure if one component can handle all the things we want in the future. For one thing the functionality will be different between each. Breasts will be drinkable but cock not. Cocks will have the cum verb attached which allows them to eject special content.

**Comment by Memeji** [comment]
Lemme know how you'd like to move forward with this. As well as how I would make those changes to this PR!

**Comment by FoxxoTrystan** [comment]
Il give it a proper review tomorrow.

**Review by FoxxoTrystan — CHANGES_REQUESTED** [review]
HasBoobsSystem and HasPenisSystem just compleatly does the same thing, this need to be regrouped into one system, by using multiple component or .yml files.

Im willing to assist or make some commits if you need asisstance.

Theres also some functions that are outated but i will point those issues out later after the main issue is fixed.

**Review by Fansana — APPROVED** [review]

**Review by FoxxoTrystan — CHANGES_REQUESTED** [review]
I fixed merge conflicts but now test are failing due to some component not being valid, this need to be fixed.

File: Content.Server/FloofStation/Traits/LewdTraitSystem.cs [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+    //    solutionSquirt.MaxVolume = entity.Comp.MaxVolume;
+
+    //    solutionSquirt.AddReagent(entity.Comp.ReagentId, entity.Comp.MaxVolume - solutionSquirt.Volume);
+    //}
+
+    public void AddCumVerb(Entity<CumProducerComponent> entity, ref GetVerbsEvent<InnateVerb> args)
+    {
+        if (args.Using == null ||
+             !args.CanInteract ||
+             !EntityManager.HasComponent<RefillableSolutionComponent>(args.Using.Value)) //see if removing this part lets you milk on the ground.
```
Comment (FoxxoTrystan) [inline Content.Server/FloofStation/Traits/LewdTraitSystem.cs:76]:
```suggestion
             !args.CanInteract ||
             args.User != args.Target ||
             !EntityManager.HasComponent<RefillableSolutionComponent>(args.Using.Value)) //see if removing this part lets you milk on the ground.
```

File: Content.Server/FloofStation/Traits/LewdTraitSystem.cs [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+            Priority = 1
+        };
+        args.Verbs.Add(verbCum);
+    }
+
+    public void AddMilkVerb(Entity<MilkProducerComponent> entity, ref GetVerbsEvent<InnateVerb> args)
+    {
+        if (args.Using == null ||
+             !args.CanInteract ||
+             !EntityManager.HasComponent<RefillableSolutionComponent>(args.Using.Value)) //see if removing this part lets you milk on the ground.
```
Comment (FoxxoTrystan) [inline Content.Server/FloofStation/Traits/LewdTraitSystem.cs:97]:
```suggestion
             !args.CanInteract ||
             args.User != args.Target ||
             !EntityManager.HasComponent<RefillableSolutionComponent>(args.Using.Value)) //see if removing this part lets you milk on the ground.
```

**Review by FoxxoTrystan — CHANGES_REQUESTED** [review]

**Review by FoxxoTrystan — APPROVED** [review]
Honestly, this will need updates, refractors, ect... to be improved but so far, right now.

This is fine.

**Comment by Leviathan21** [comment]
every dya we strafe further from god

**Pull request merged**